pkg/discovery/discovery.go

```diff
@@ -28,7 +28,7 @@ type Backend interface {
 	Watcher
 
 	// Initialize the discovery with URIs, a heartbeat and a ttl.
-	Initialize(string, time.Duration, time.Duration) error
+	Initialize(string, time.Duration, time.Duration, map[string]string) error
```

Comment:
new `map[string]string` param needs a comment describing its purpose, like the other params.